### PR TITLE
Add typescript typings definition file

### PR DESCRIPTION
**Pull request by alex3165**
Tested with `Liveprovider` and `LiveEditor` but it should be alright.

### CODE REVIEW

File: react-live.d.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+export interface EditorProps extends HTMLElement {
+  onChange: Function;
+  onKeyDown: Function;
+  onKeyUp: Function;
+  onClick: Function;
+  code: string;
+}
+
+export function LiveEditor(props: EditorProps): any
```
Comment (philpl) [inline react-live.d.ts]: Shouldn't those be typed as StatelessFunctionalComponents? :)

File: react-live.d.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+export function LiveError(props: HTMLElement): any
+export function LivePreview(props: HTMLElement): any
+
+export class Editor extends Component<HTMLElement, {}>{}
+
+export function withLive(wrappedComponent: Component<any, any>): Component<any, {}>
```
Comment (philpl) [inline react-live.d.ts]: Component probably doesn't accept SFCs :)

File: react-live.d.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+export class LiveProvider extends Component<LiveProviderProps, {}>{}
+
+export interface EditorProps extends HTMLElement {
+  onChange: Function;
+  onKeyDown: Function;
+  onKeyUp: Function;
+  onClick: Function;
+  code: string;
+}
+
```
Comment (philpl) [inline react-live.d.ts]: I think the normal `Editor` is missing :) https://github.com/FormidableLabs/react-live/blob/master/src/index.js#L13

File: react-live.d.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  onError?: Function;
+  onChange?: Function;
+  element?: any;
+}
+
+export class LiveProvider extends Component<LiveProviderProps, {}>{}
+
+export interface EditorProps extends HTMLElement {
+  onChange: Function;
+  onKeyDown: Function;
```
Comment (philpl) [inline react-live.d.ts]: Wasn't there some kind of special function type for events?

File: react-live.d.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  code: string;
+  error?: string;
+  onError?: Function;
+  onChange?: Function;
+  element?: any;
+}
+
+export class LiveProvider extends Component<LiveProviderProps, {}>{}
+
+export interface EditorProps extends HTMLElement {
```
Comment (philpl) [inline react-live.d.ts]: LiveEditor also accepts the props of Editor, so I don't know if `HTMLElement` includes `contentEditable` ^^

File: react-live.d.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
+export as namespace ReactLive;
+
+import { Component } from 'react'
+
+export interface LiveProviderProps {
```
Comment (philpl) [inline react-live.d.ts]: These are all the context types actually ;)

LiveProvider itself accepts: 

```
static propTypes = {
    className: PropTypes.string,
    code: PropTypes.string,
    scope: PropTypes.object,
    mountStylesheet: PropTypes.bool,
    noInline: PropTypes.bool
  }
```

File: react-live.d.ts (hunk 5 above)
Comment (alex3165) [inline react-live.d.ts]: it does I have tested it

File: react-live.d.ts (hunk 6 above)
Comment (alex3165) [inline react-live.d.ts]: Ah yeah my mistake ahah

File: react-live.d.ts (hunk 4 above)
Comment (alex3165) [inline react-live.d.ts]: done

File: react-live.d.ts (hunk 3 above)
Comment (alex3165) [inline react-live.d.ts]: Yes editor is there: `export class Editor extends Component<HTMLElement, {}>{}`

File: react-live.d.ts (hunk 1 above)
Comment (alex3165) [inline react-live.d.ts]: Well function works as well apparently?

File: react-live.d.ts (hunk 2 above)
Comment (alex3165) [inline react-live.d.ts]: done

File: react-live.d.ts [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+export function LiveEditor(props: EditorProps): any
+
+export function LiveError(props: HTMLElement): any
+export function LivePreview(props: HTMLElement): any
+
+export class Editor extends Component<HTMLElement, {}>{}
+
+export function withLive(wrappedComponent: Component<any, any> | StatelessComponent<any>): Component<any, {}>
```
Comment (philpl) [inline react-live.d.ts]: Can we put in a generic here? :)

Also, this should attach the `live` prop with the context properties

File: react-live.d.ts [hunk 8]
```diff
@@ -0,0 +1,30 @@
+export as namespace ReactLive;
+
+import { Component, StatelessComponent } from 'react'
+
+export interface LiveProviderProps {
```
Comment (philpl) [inline react-live.d.ts]: This can also just extend the HTMLElement props

File: react-live.d.ts (hunk 1 above)
Comment (philpl) [inline react-live.d.ts]: Yea, but the return type will be wrong ^^

**Comment by alex3165** [comment]
Should be all good now 👍

**Pull request merged**